src/NGen.cpp

```diff
@@ -259,11 +263,29 @@ int main(int argc, char *argv[]) {
     #endif // NGEN_MPI_ACTIVE
 
     // TODO: Instead of iterating through a collection of FeatureBase objects mapping to nexi, we instead want to iterate through HY_HydroLocation objects
-    geojson::GeoJSON nexus_collection = geojson::read(nexusDataFile, nexus_subset_ids);
+    geojson::GeoJSON nexus_collection;
+    if (boost::algorithm::ends_with(nexusDataFile, "gpkg")) {
+      #ifdef NGEN_WITH_SQLITE3
+      nexus_collection = geopackage::read(nexusDataFile, "nexus", nexus_subset_ids);
+      #else
+      throw std::runtime_error("SQLite3 support required to read GeoPackage files.");
+      #endif
+    } else {
+      nexus_collection = geojson::read(nexusDataFile, nexus_subset_ids);
+    }
     std::cout << "Building Catchment collection" << std::endl;
 
     // TODO: Instead of iterating through a collection of FeatureBase objects mapping to catchments, we instead want to iterate through HY_Catchment objects
-    geojson::GeoJSON catchment_collection = geojson::read(catchmentDataFile, catchment_subset_ids);
+    geojson::GeoJSON catchment_collection;
+    if (boost::algorithm::ends_with(catchmentDataFile, "gpkg")) {
```

Comment:
This is a temporary "workaround", my reasoning was that I did not want to immediately change the cli interface, so the intent wasn't directly to be able to support both GPKG and GeoJSON files on the same run. I imagine once we get around to refactoring the cli, this would be handled so that just the single GPKG path is passed.

Reply:
Indeed.... either the CLI or config file refactor is needed to "fix" this. The HF (or the individual components of it) might belong in the realization config in a multi-layer world, because (at least eventually) different layers will use other hydrofabrics or discretizations that don't really need a hydrofabric for definition... So while the HF discretization would probably be a special case used for multiprocessing division, it would be just one of possibly many discretizations specified within a simulation.